Dining/views.py

```diff
@@ -239,11 +259,12 @@ def get(self, request, *args, **kwargs):
         return self.render_to_response(context)
 
     def post(self, request, *args, **kwargs):
-        # Todo: re-enable External Dining Entries
         context = self.get_context_data()
 
+        # Check permissions, if user has no access to this page, reject it.
+        if not self.check_user_permission(request, request.user):
+            return HttpResponseRedirect(self.reverse('slot_details'))
```

Comment:
No 401 or some kind of warning message?

Reply:
The check_user_permissions generates warning messages. I didn't do a 401 as that doesn't help user experience imo as it basically signals a dead-end while the reason is not because the page is normally blocked, only in this specific instance (dininglist is already closed or full, i.e. an hour earlier the page was availlable)